src/Cache/CacheEngine.php

```diff
@@ -97,7 +97,7 @@ public function init(array $config = []): bool
      * @return void
      * @throws \Cake\Cache\InvalidArgumentException When the key is not valid.
      */
-    protected function ensureValidKey($key): void
+    protected function ensureValidKey(string $key): void
```

Comment:
This needs to be `mixed` as it's actually a validation function.

Reply:
You're right. I was just looking at this error.